apis/iam/v1beta1/diff.go

```diff
@@ -53,23 +53,20 @@ func IAMPolicySpecDiffers(desired, actual *IAMPolicySpec) *structuredreporting.D
 // It treats the slices as maps keyed by Role, and members are treated as unordered sets.
 func compareBindings(desired, actual []IAMPolicyBinding) *structuredreporting.Diff {
 	diff := &structuredreporting.Diff{}
-	desiredMap := make(map[string]IAMPolicyBinding)
-	for _, b := range desired {
-		sort.Slice(b.Members, func(i, j int) bool {
-			return b.Members[i] < b.Members[j]
-		})
-		// Canonicalize members for stable comparison.
-		desiredMap[b.Role] = b
-	}
 
-	actualMap := make(map[string]IAMPolicyBinding)
-	for _, b := range actual {
-		sort.Slice(b.Members, func(i, j int) bool {
-			return b.Members[i] < b.Members[j]
-		})
-		// Canonicalize members for stable comparison.
-		actualMap[b.Role] = b
+	deepCopySortBindings := func(bindings []IAMPolicyBinding) map[string]IAMPolicyBinding {
+		m := make(map[string]IAMPolicyBinding)
+		for _, b := range bindings {
+			bindingCopy := *b.DeepCopy()
```

Comment:
Similarly, `auditConfigsToMap` assumes that the `Service` field is a unique key. If duplicate services can exist in the `AuditConfigs` slice, this could lead to incorrect diffs.